helper/resource/testing_test.go

```diff
@@ -509,22 +509,22 @@ func TestTest_stepError(t *testing.T) {
 
 func TestTest_factoryError(t *testing.T) {
 	resourceFactoryError := fmt.Errorf("resource factory error")
-
 	factory := func() (terraform.ResourceProvider, error) {
 		return nil, resourceFactoryError
 	}
-
 	mt := new(mockT)
-	Test(mt, TestCase{
-		ProviderFactories: map[string]terraform.ResourceProviderFactory{
-			"test": factory,
-		},
-		Steps: []TestStep{
-			{
-				ExpectError: regexp.MustCompile("resource factory error"),
+
+	func() {
+		defer func() {
+			recover()
```

Comment:
just to sanity check here, i would ensure the `recover` value is `"mockT.Fatal"` or something and not an unrelated panic.